In [ ]:
import pandas as pd

#Reading the Dataset

In [ ]:
df = pd.read_csv("water_potability.csv")
df.head()

ph    Hardness        Solids  Chloramines     Sulfate  Conductivity  \
0       NaN  204.890455  20791.318981     7.300212  368.516441    564.308654   
1  3.716080  129.422921  18630.057858     6.635246         NaN    592.885359   
2  8.099124  224.236259  19909.541732     9.275884         NaN    418.606213   
3  8.316766  214.373394  22018.417441     8.059332  356.886136    363.266516   
4  9.092223  181.101509  17978.986339     6.546600  310.135738    398.410813   

   Organic_carbon  Trihalomethanes  Turbidity  Potability  
0       10.379783        86.990970   2.963135           0  
1       15.180013        56.329076   4.500656           0  
2       16.868637        66.420093   3.055934           0  
3       18.436524       100.341674   4.628771           0  
4       11.558279        31.997993   4.075075           0

#Import library for Naive Bayes

In [ ]:
import numpy as np # linear algebra
import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)
import matplotlib.pyplot as plt # for data visualization purposes
import seaborn as sns # for statistical data visualization


#Gaussian Model

#Informations of the dataset

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3276 entries, 0 to 3275
Data columns (total 10 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   ph               2785 non-null   float64
 1   Hardness         3276 non-null   float64
 2   Solids           3276 non-null   float64
 3   Chloramines      3276 non-null   float64
 4   Sulfate          2495 non-null   float64
 5   Conductivity     3276 non-null   float64
 6   Organic_carbon   3276 non-null   float64
 7   Trihalomethanes  3114 non-null   float64
 8   Turbidity        3276 non-null   float64
 9   Potability       3276 non-null   int64  
dtypes: float64(9), int64(1)
memory usage: 256.1 KB


#Drop NaN values

In [ ]:
df = df.dropna().reset_index(drop=True)
df

ph    Hardness        Solids  Chloramines     Sulfate  \
0      8.316766  214.373394  22018.417441     8.059332  356.886136   
1      9.092223  181.101509  17978.986339     6.546600  310.135738   
2      5.584087  188.313324  28748.687739     7.544869  326.678363   
3     10.223862  248.071735  28749.716544     7.513408  393.663396   
4      8.635849  203.361523  13672.091764     4.563009  303.309771   
...         ...         ...           ...          ...         ...   
2006   8.989900  215.047358  15921.412018     6.297312  312.931022   
2007   6.702547  207.321086  17246.920347     7.708117  304.510230   
2008  11.491011   94.812545  37188.826022     9.263166  258.930600   
2009   6.069616  186.659040  26138.780191     7.747547  345.700257   
2010   4.668102  193.681735  47580.991603     7.166639  359.948574   

      Conductivity  Organic_carbon  Trihalomethanes  Turbidity  Potability  
0       363.266516       18.436524       100.341674   4.628771           0  
1       398.410813       11.558279        31.997993   4.075075           0  
2       280.467916        8.399735        54.917862   2.559708           0  
3       283.651634       13.789695        84.603556   2.672989           0  
4       474.607645       12.363817        62.798309   4.401425           0  
...            ...             ...              ...        ...         ...  
2006    390.410231        9.899115        55.069304   4.613843           1  
2007    329.266002       16.217303        28.878601   3.442983           1  
2008    439.893618       16.172755        41.558501   4.369264           1  
2009    415.886955       12.067620        60.419921   3.669712           1  
2010    526.424171       13.894419        66.687695   4.435821           1  

[2011 rows x 10 columns]

#Declare feature vector and target variable

In [ ]:
X = df.drop(['Potability'], axis=1)
y = df['Potability']


#Split data into separate training and test set 

In [ ]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.3, random_state = 0)

In [ ]:
# check the shape of X_train and X_test

X_train.shape, X_test.shape

((1407, 9), (604, 9))

#Feature Scaling 

In [ ]:
cols = X_train.columns

In [ ]:
from sklearn.preprocessing import RobustScaler

scaler = RobustScaler()

X_train = scaler.fit_transform(X_train)

X_test = scaler.transform(X_test)

In [ ]:
X_train = pd.DataFrame(X_train, columns=[cols])

In [ ]:
X_test = pd.DataFrame(X_test, columns=[cols])

In [ ]:
X_train.head()

ph  Hardness    Solids Chloramines   Sulfate Conductivity  \
0  0.433417  2.001473  0.530563    1.067017  1.772637    -0.556168   
1 -1.632827 -0.190031 -0.216287   -0.113595  1.221826    -0.499500   
2  0.662243 -0.440240  0.084853   -0.960733  0.134540    -0.225171   
3 -0.736981 -0.222374  0.664751    0.218844 -0.111549    -1.239846   
4 -0.067825  0.645418 -0.273850    0.447193 -0.635374     0.056382   

  Organic_carbon Trihalomethanes Turbidity  
0       0.181255       -0.456707 -0.271335  
1       0.228745        0.946304  0.831617  
2      -0.122746        2.062952 -1.091632  
3      -1.309287       -0.547060 -1.304534  
4      -0.451152        0.107355 -1.180735

#Model Training

In [ ]:
# train a Gaussian Naive Bayes classifier on the training set
from sklearn.naive_bayes import GaussianNB


# instantiate the model
gnb = GaussianNB()


# fit the model
gnb.fit(X_train, y_train)

/usr/local/lib/python3.7/dist-packages/sklearn/utils/validation.py:1692: FutureWarning: Feature names only support names that are all strings. Got feature names with dtypes: ['tuple']. An error will be raised in 1.2.
  FutureWarning,


GaussianNB()

#Predict the results 

In [ ]:
y_pred = gnb.predict(X_test)

y_pred

/usr/local/lib/python3.7/dist-packages/sklearn/utils/validation.py:1692: FutureWarning: Feature names only support names that are all strings. Got feature names with dtypes: ['tuple']. An error will be raised in 1.2.
  FutureWarning,


array([0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 1, 0, 1,
       0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0,
       0, 0, 1, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0,
       1, 0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       1, 0, 0, 0, 1, 0, 0, 0, 0, 1, 1, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0,
       0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1,
       0, 0, 1, 0, 0, 0, 1, 1, 1, 0, 0, 1, 0, 1, 0,

#Accuracy of the model

In [ ]:
from sklearn.metrics import accuracy_score

print('Model accuracy score: {0:0.2f}'. format(accuracy_score(y_test, y_pred) *100))

Model accuracy score: 62.91


Compare the train-set and test-set accuracy.

Now, I will compare the train-set and test-set accuracy to check for overfitting.

In [ ]:
y_pred_train = gnb.predict(X_train)

y_pred_train


/usr/local/lib/python3.7/dist-packages/sklearn/utils/validation.py:1692: FutureWarning: Feature names only support names that are all strings. Got feature names with dtypes: ['tuple']. An error will be raised in 1.2.
  FutureWarning,


array([1, 0, 0, ..., 0, 0, 1])

In [ ]:
print('Training-set accuracy score: {0:0.2f}'. format(accuracy_score(y_train, y_pred_train) *100))

Training-set accuracy score: 63.11


Check for overfitting and underfitting

In [ ]:
print('Training set score: {:.2f}'.format(gnb.score(X_train, y_train) *100))

print('Test set score: {:.2f}'.format(gnb.score(X_test, y_test)*100))

Training set score: 63.11
Test set score: 62.91


/usr/local/lib/python3.7/dist-packages/sklearn/utils/validation.py:1692: FutureWarning: Feature names only support names that are all strings. Got feature names with dtypes: ['tuple']. An error will be raised in 1.2.
  FutureWarning,
/usr/local/lib/python3.7/dist-packages/sklearn/utils/validation.py:1692: FutureWarning: Feature names only support names that are all strings. Got feature names with dtypes: ['tuple']. An error will be raised in 1.2.
  FutureWarning,


The values are really close, so there is no sign of overfitting

Overfitting: In train the model has good result but with test, he has a bad result

Underfitting: The model is bad in the trainig already, so there is no need of the test ocurrs.

#Compare model accuracy with null accuracy

In [ ]:
y_test.value_counts()

0    367
1    237
Name: Potability, dtype: int64

In [ ]:
null_accuracy = (367/(367+237))

print('Null accuracy score: {0:0.2f}'. format(null_accuracy *100))

Null accuracy score: 60.76


So the accuracy of our Gaussian Naive Bayes Classification model it's very good beacaus the null model it's 60.76% and our model make an accuracy of 62.91%, so we are doing a good job

#Results

In [ ]:
# Importing metrics of validation
from sklearn.metrics import classification_report, confusion_matrix, accuracy_score

# Confuse Matrix
print("Confusion Matrix: ")
print(confusion_matrix(y_test, y_pred), "\n")  

# Classification Report
print("Classification Report: \n", classification_report(y_test, y_pred))  


# 
print('Model of the Accuracy: ' , (accuracy_score(y_test, y_pred) *100))

Confusion Matrix: 
[[328  39]
 [185  52]] 

Classification Report: 
               precision    recall  f1-score   support

           0       0.64      0.89      0.75       367
           1       0.57      0.22      0.32       237

    accuracy                           0.63       604
   macro avg       0.61      0.56      0.53       604
weighted avg       0.61      0.63      0.58       604

Model of the Accuracy:  62.913907284768214


#Multinomial

#Splite data in test and train

In [ ]:
X = df.drop(['Potability'], axis=1)
y = df['Potability']


In [ ]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.3, random_state = 0)

In [ ]:
X_train.shape, X_test.shape

((1407, 9), (604, 9))

#Model Training

In [ ]:
from sklearn.naive_bayes import  MultinomialNB

In [ ]:
# instantiate the model
mlt =  MultinomialNB()


# fit the model
mlt.fit(X_train, y_train)

MultinomialNB()

#Predict the results

In [ ]:
y_pred = mlt.predict(X_test)

y_pred

array([1, 1, 0, 1, 0, 1, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 0, 1, 1, 1, 1,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 0, 1, 0, 1, 0,
       1, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 1, 0, 0, 1, 0, 0, 0, 1, 1, 1,
       0, 0, 1, 0, 1, 0, 0, 1, 0, 0, 0, 1, 1, 1, 0, 1, 0, 1, 0, 0, 1, 0,
       1, 0, 0, 1, 0, 0, 1, 0, 1, 1, 1, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 1,
       0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 1, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0, 1,
       0, 1, 1, 1, 1, 1, 0, 0, 1, 1, 1, 0, 0, 1, 1, 0, 0, 0, 1, 0, 1, 0,
       0, 1, 1, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 1, 0,
       0, 1, 0, 1, 1, 0, 0, 0, 1, 0, 1, 1, 1, 0, 1, 0, 1, 0, 1, 0, 0, 0,
       0, 1, 0, 0, 1, 1, 1, 1, 0, 1, 0, 1, 0, 0, 0, 0, 1, 1, 1, 0, 1, 0,
       1, 1, 0, 1, 1, 0, 0, 1, 1, 1, 1, 0, 1, 0, 1, 0, 0, 1, 0, 1, 0, 1,
       1, 0, 1, 1, 1, 1, 1, 1, 0, 1, 1, 0, 1, 0, 0, 1, 1, 0, 0, 1, 1, 0,
       0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1, 1, 1, 0, 0, 1, 0, 0, 0, 0, 1, 1,
       1, 1, 0, 1, 0, 0, 0, 1, 1, 0, 0, 1, 0, 1, 1,

#Accuracy of the model

In [ ]:
from sklearn.metrics import accuracy_score

print('Model accuracy score: {0:0.2f}'. format(accuracy_score(y_test, y_pred) *100))

Model accuracy score: 52.32


Compare the train-set and test-set accuracy.

Now, I will compare the train-set and test-set accuracy to check for overfitting.

In [ ]:
y_pred_train = mlt.predict(X_train)

y_pred_train

array([1, 0, 0, ..., 1, 0, 1])

In [ ]:
print('Training-set accuracy score: {0:0.2f}'. format(accuracy_score(y_train, y_pred_train) *100))

Training-set accuracy score: 52.45


Check for overfitting and underfitting

In [ ]:
print('Training set score: {:.2f}'.format(mlt.score(X_train, y_train) *100))

print('Test set score: {:.2f}'.format(mlt.score(X_test, y_test)*100))

Training set score: 52.45
Test set score: 52.32


The values are really close, so there is no sign of overfitting

Overfitting: In train the model has good result but with test, he has a bad result

Underfitting: The model is bad in the trainig already, so there is no need of the test ocurrs.

#Compare model accuracy with null accuracy

In [ ]:
y_test.value_counts()


0    367
1    237
Name: Potability, dtype: int64

In [ ]:
null_accuracy = (367/(367+237))

print('Null accuracy score: {0:0.2f}'. format(null_accuracy *100))

Null accuracy score: 60.76


So the accuracy of our Multidimensional Naive Bayes Classification model it's very good beacaus the null model it's 60.76% and our model make an accuracy of 52.32%, so we are doing a good job, but wors tahn gaussian model

#Results

In [ ]:
# Importing metrics of validation
from sklearn.metrics import classification_report, confusion_matrix, accuracy_score

# Confuse Matrix
print("Confusion Matrix: ")
print(confusion_matrix(y_test, y_pred), "\n")  

# Classification Report
print("Classification Report: \n", classification_report(y_test, y_pred))  


# 
print('Model of the Accuracy: ' , (accuracy_score(y_test, y_pred) *100))

Confusion Matrix: 
[[212 155]
 [133 104]] 

Classification Report: 
               precision    recall  f1-score   support

           0       0.61      0.58      0.60       367
           1       0.40      0.44      0.42       237

    accuracy                           0.52       604
   macro avg       0.51      0.51      0.51       604
weighted avg       0.53      0.52      0.53       604

Model of the Accuracy:  52.317880794701985


#Bernoulli

#Splite data in test and train

In [ ]:
X = df.drop(['Potability'], axis=1)
y = df['Potability']


In [ ]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.3, random_state = 0)

In [ ]:
X_train.shape, X_test.shape

((1407, 9), (604, 9))

#Feature Scaling

In [ ]:
cols = X_train.columns

In [ ]:
from sklearn.preprocessing import RobustScaler

scaler = RobustScaler()

X_train = scaler.fit_transform(X_train)

X_test = scaler.transform(X_test)

In [ ]:
X_train = pd.DataFrame(X_train, columns=[cols])

In [ ]:
X_test = pd.DataFrame(X_test, columns=[cols])

In [ ]:
X_train.head()

ph  Hardness    Solids Chloramines   Sulfate Conductivity  \
0  0.433417  2.001473  0.530563    1.067017  1.772637    -0.556168   
1 -1.632827 -0.190031 -0.216287   -0.113595  1.221826    -0.499500   
2  0.662243 -0.440240  0.084853   -0.960733  0.134540    -0.225171   
3 -0.736981 -0.222374  0.664751    0.218844 -0.111549    -1.239846   
4 -0.067825  0.645418 -0.273850    0.447193 -0.635374     0.056382   

  Organic_carbon Trihalomethanes Turbidity  
0       0.181255       -0.456707 -0.271335  
1       0.228745        0.946304  0.831617  
2      -0.122746        2.062952 -1.091632  
3      -1.309287       -0.547060 -1.304534  
4      -0.451152        0.107355 -1.180735

#Model Training

In [ ]:
from sklearn.naive_bayes import BernoulliNB

In [ ]:
# instantiate the model
ber =  BernoulliNB()


# fit the model
ber.fit(X_train, y_train)

/usr/local/lib/python3.7/dist-packages/sklearn/utils/validation.py:1692: FutureWarning: Feature names only support names that are all strings. Got feature names with dtypes: ['tuple']. An error will be raised in 1.2.
  FutureWarning,


BernoulliNB()

#Predict the results

In [ ]:
y_pred = ber.predict(X_test)

y_pred

/usr/local/lib/python3.7/dist-packages/sklearn/utils/validation.py:1692: FutureWarning: Feature names only support names that are all strings. Got feature names with dtypes: ['tuple']. An error will be raised in 1.2.
  FutureWarning,


array([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
       0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,

#Accuracy of the model

In [ ]:
from sklearn.metrics import accuracy_score

print('Model accuracy score: {0:0.2f}'. format(accuracy_score(y_test, y_pred) *100))

Model accuracy score: 60.76


Compare the train-set and test-set accuracy.

Now, I will compare the train-set and test-set accuracy to check for overfitting.

In [ ]:
y_pred_train = ber.predict(X_train)

y_pred_train

/usr/local/lib/python3.7/dist-packages/sklearn/utils/validation.py:1692: FutureWarning: Feature names only support names that are all strings. Got feature names with dtypes: ['tuple']. An error will be raised in 1.2.
  FutureWarning,


array([0, 0, 0, ..., 0, 0, 0])

In [ ]:
print('Training-set accuracy score: {0:0.2f}'. format(accuracy_score(y_train, y_pred_train) *100))

Training-set accuracy score: 59.20


Check for overfitting and underfitting

In [ ]:
print('Training set score: {:.2f}'.format(ber.score(X_train, y_train) *100))

print('Test set score: {:.2f}'.format(ber.score(X_test, y_test)*100))

Training set score: 59.20
Test set score: 60.76


/usr/local/lib/python3.7/dist-packages/sklearn/utils/validation.py:1692: FutureWarning: Feature names only support names that are all strings. Got feature names with dtypes: ['tuple']. An error will be raised in 1.2.
  FutureWarning,
/usr/local/lib/python3.7/dist-packages/sklearn/utils/validation.py:1692: FutureWarning: Feature names only support names that are all strings. Got feature names with dtypes: ['tuple']. An error will be raised in 1.2.
  FutureWarning,


The values are really close, so there is no sign of overfitting

Overfitting: In train the model has good result but with test, he has a bad result

Underfitting: The model is bad in the trainig already, so there is no need of the test ocurrs.

#Compare model accuracy with null accuracy

In [ ]:
y_test.value_counts()

0    367
1    237
Name: Potability, dtype: int64

In [ ]:
null_accuracy = (367/(367+237))

print('Null accuracy score: {0:0.2f}'. format(null_accuracy *100))


Null accuracy score: 60.76


So the accuracy of ourBernouli Naive Bayes Classification model it's very good because the have the same value of the null values.

Referencias:
1. https://www.kaggle.com/code/prashant111/naive-bayes-classifier-in-python/notebook

2. https://scikit-learn.org/stable/modules/generated/sklearn.naive_bayes.BernoulliNB.html

3. https://didatica.tech/underfitting-e-overfitting/

4. https://machinelearningmastery.com/precision-recall-and-f-measure-for-imbalanced-classification/

#Results

In [ ]:
# Importing metrics of validation
from sklearn.metrics import classification_report, confusion_matrix, accuracy_score

# Confuse Matrix
print("Confusion Matrix: ")
print(confusion_matrix(y_test, y_pred), "\n")  

# Classification Report
print("Classification Report: \n", classification_report(y_test, y_pred))  


# 
print('Model of the Accuracy: ' , (accuracy_score(y_test, y_pred) *100))

/usr/local/lib/python3.7/dist-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
/usr/local/lib/python3.7/dist-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
/usr/local/lib/python3.7/dist-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


Confusion Matrix: 
[[367   0]
 [237   0]] 

Classification Report: 
               precision    recall  f1-score   support

           0       0.61      1.00      0.76       367
           1       0.00      0.00      0.00       237

    accuracy                           0.61       604
   macro avg       0.30      0.50      0.38       604
weighted avg       0.37      0.61      0.46       604

Model of the Accuracy:  60.76158940397352


In [ ]:
from sklearn.metrics import recall_score

Recall Bernoli

In [ ]:
recall = recall_score(y_test, y_pred, average='binary')
print('Recall: %.2f' % (recall*100))

Recall: 0.00


Recall Gaussian

In [ ]:
recall = recall_score(y_test, y_pred, average='binary')
print('Recall: %.2f' % (recall*100))

Recall: 0.00


Recall Multinomial

In [ ]:
recall = recall_score(y_test, y_pred, average='binary')
print('Recall: %.2f' % (recall*100))

Recall: 0.00
